zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZookeeperRegisterBuilder.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.zookeeper;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.zookeeper.NodeValueCodec;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerListener;
+import com.linecorp.armeria.server.ServerListenerBuilder;
+
+/**
+ * Builds a new {@link ServerListener}, which registers the server to the zookeeper instance.
+ * <h2>Example</h2>
+ * <pre>{@code
+ * ServerListener register = new ZookeeperRegisterBuilder()
+ * // Zookeeper Connection String
+ * .connect("myZooKeeperHost:2181")
+ * // Zookeeper Node to use
+ * .node("/myProductionEndpoints")
+ * // Session Timeout
+ * .sessionTimeOut(10000)
+ * // NodeValueCodec
+ * .nodeValueCodec(NodeValueCodec.DEFAULT)
+ * .build();
+ * // Set to `Server`.
+ * Server server = ...
+ * server.serverListener(register);
+ * }</pre>
+ * */
+public class ZookeeperRegisterBuilder {
+    private static final int DEFAULT_SESSION_TIMEOUT = 10000;
+
+    private String connect;
+    private String node;
+    private int sessionTimeout = DEFAULT_SESSION_TIMEOUT;
+    private Endpoint endpoint;
+    private NodeValueCodec nodeValueCodec = NodeValueCodec.DEFAULT;
+
+    /**
+     * Sets the zookeeper connection string.
+     *
+     * @param connect the zookeeper connection string.
+     */
+    public ZookeeperRegisterBuilder connect(String connect) {
+        checkArgument(!isNullOrEmpty(connect), "connect");
+        this.connect = connect;
+        return this;
+    }
+
+    /**
+     * Sets the zookeeper node to register.
+     *
+     * @param node the zookeeper node to register.
+     */
+    public ZookeeperRegisterBuilder node(String node) {
+        checkArgument(!isNullOrEmpty(node), "node");
+        this.node = node;
+        return this;
+    }
+
+    /**
+     * Sets the session timeout (in ms). default is 10000.
+     *
+     * @param timeout the session timeout.
+     */
+    public ZookeeperRegisterBuilder sessionTimeout(int timeout) {
```

Comment:
- `sessionTimeoutMillis(long timeoutMillis)` (Let's convert it into `int` internally.)
- Please also add `sessionTimeout(Duration)` as well.